### PR TITLE
[Mailbox]: Added the ability to mark as read/unread for a single thread + bulk actions

### PR DESCRIPTION
**Pull request by pooja169usp**
- Added `updateThread` method to threads store
- Calling  `updateThread` method through MailboxStore in Mailbox component for both single thread & multiple threads status change of read/unread 

## Side-effect:
- Fixed show participants logic
- Fixed show participants style where the plus sign & additional participants count occupied 2 lines 

# Readiness checklist

- [X] Cypress tests passing?

# License

I confirm that this contribution is made under the terms of the MIT license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

**Review by wonka-lighthouse — COMMENTED** [review]
| Device | URL |
|--|--|
| mobile | http://localhost:8000/components/agenda/src/index.html |

     
| Device | URL |
|--|--|
| mobile | http://localhost:8000/components/composer/src/index.html |

Not what you expected? Are your scores flaky? [Run Lighthouse on Foo](https://www.foo.software/lighthouse)
If scores continue to be inconsistent consider [running all audits on Foo](https://github.com/foo-software/lighthouse-check-action#usage-foos-automated-lighthouse-check-api)

File: commons/src/store/threads.ts [hunk 1]
```diff
@@ -31,7 +36,20 @@ function initializeThreads() {
       });
       return threadsMap[queryKey];
     },
-
+    updateThread: async (
+      threadQuery: ConversationQuery,
+      queryKey: string,
+      updatedThread: Conversation,
+    ) => {
+      await updateThread(threadQuery, updatedThread).then((thread) => {
```
Comment (ozsivanov) [inline commons/src/store/threads.ts:44]: nit - This can be refactored this way to avoid the nested code block
```suggestion
      const thread = await updateThread(threadQuery, updatedThread);
```

File: commons/src/store/threads.ts [hunk 2]
```diff
@@ -31,7 +36,20 @@ function initializeThreads() {
...
+        threadsMap[queryKey] = threadsMap[queryKey].map((_thread) => {
```
Comment (ozsivanov) [inline commons/src/store/threads.ts:45]: nit - Consider renaming `_thread`, it makes the logic below a bit hard to follow. Or maybe this var can be `thread` and the current thread can be renamed to `updatedThread` to avoid confusion between the two.

**Review by ozsivanov — APPROVED** [review]

**Review by philrenaud — APPROVED** [review]
